theano/gpuarray/c_code/spatialtf_grid.c

```diff
@@ -0,0 +1,86 @@
+#section support_code
+
+int
+spatialtf_grid(PyGpuArrayObject * theta,
+               PyArrayObject * grid_dimensions,
+               cudnnSpatialTransformerDescriptor_t desc,
+               PyGpuArrayObject ** grid,
+               cudnnHandle_t _handle)
+{
+    PyGpuContextObject * gpu_ctx = theta->context;
+    cudnnStatus_t err;
+
+    if ( theta->ga.typecode != GA_FLOAT &&
+         theta->ga.typecode != GA_DOUBLE &&
+         theta->ga.typecode != GA_HALF )
+    {
+        PyErr_SetString( PyExc_TypeError, "Unsupported data type for theta" );
+        return -1;
+    }
+
+    if ( PyGpuArray_NDIM( theta ) != 3 )
+    {
+        PyErr_Format( PyExc_RuntimeError,
+                      "theta must have three dimensions!" );
+        return -1;
+    }
+
+    if ( PyGpuArray_DIM( theta, 1 ) != 2 && PyGpuArray_DIM( theta, 2 ) != 3 )
+    {
+        PyErr_Format( PyExc_RuntimeError,
+                      "Incorrect dimensions for theta, should be (%d, %d, %d), got (%d, %d, %d)",
+                      PyGpuArray_DIMS( theta )[0], 2, 3, PyGpuArray_DIMS( theta )[0],
+                      PyGpuArray_DIMS( theta )[1], PyGpuArray_DIMS( theta )[2] );
+        return -1;
+    }
+
+    if ( PyArray_DIM( grid_dimensions, 0 ) < 3 )
```

Comment:
Grid dimensions should be at least 3D (number of images, height, width), since gray-scale images do not have the number of channels dimension.